DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/storm_control/test_storm_control_broadcast_traffic.py

```diff
@@ -0,0 +1,150 @@
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    traffic_duration = 10
+    pps_value = 1000
+
+    out = await IpLink.add(
+        input_data=[{device_host_name: [
+            {'device': bridge, 'vlan_filtering': 1, 'type': 'bridge'}]}])
+    err_msg = f"Verify that bridge created and vlan filtering set to 'ON'.\n{out}"
+    assert out[0][device_host_name]['rc'] == 0, err_msg
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {'device': bridge, 'operstate': 'up'}]}])
+    assert out[0][device_host_name]['rc'] == 0, f"Verify that bridge set to 'UP' state.\n{out}"
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {'device': port, 'master': bridge, 'operstate': 'up'} for port in ports]}])
+    err_msg = f"Verify that bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    assert out[0][device_host_name]['rc'] == 0, err_msg
+
+    await devlink_rate_value(dev='pci/0000:01:00.0/1', name='bc_kbyte_per_sec_rate', value=0,
+                             device_host_name=device_host_name, verify=True)
+    await devlink_rate_value(dev='pci/0000:01:00.0/1', name='bc_kbyte_per_sec_rate', value=118227, cmode='runtime',
+                             device_host_name=device_host_name, set_and_verify=True)
+
+    address_map = (
+        # swp port, tg port,    tg ip,     gw,        plen
+        (ports[0], tg_ports[0], '1.1.1.2', '1.1.1.1', 24),
+        (ports[1], tg_ports[1], '1.1.1.3', '1.1.1.1', 24)
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {'ixp': port, 'ip': ip, 'gw': gw, 'plen': plen}
+        for _, port, ip, gw, plen in address_map
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    """
+    Set up the following stream:
+    — stream_A —
+    swp1 -> swp2
+    """
+
+    streams = {
+        'stream_A': {
+            'ip_source': dev_groups[tg_ports[0]][0]['name'],
+            'ip_destination': dev_groups[tg_ports[1]][0]['name'],
+            'srcIp': '104.52.13.241',
+            'dstIp': '255.255.255.255',
+            'srcMac': '92:cc:23:09:37:ca',
+            'dstMac': 'ff:ff:ff:ff:ff:ff',
+            'frameSize': randrange(100, 1500),
+            'rate': pps_value,
+            'protocol': '0x0800',
+            'type': 'raw'
+        }
+    }
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+
+    # check the traffic stats
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, 'Traffic Item Statistics')
+    for row in stats.Rows:
+        assert tgen_utils_get_loss(row) == 0.000, \
+            f"Verify that traffic from {row['Tx Port']} to {row['Rx Port']} forwarded.\n{out}"
+
+    await cleanup_kbyte_per_sec_rate_value(dent_dev, bc=True)
```

Comment:
What if you fail at L147? Please make sure that you clean up after the test even in an exception is raised mid-test.